shared/bls/bls.go

```diff
@@ -109,10 +109,26 @@ func (s *Signature) Verify(msg []byte, pub *PublicKey, domain uint64) bool {
 	return g1.VerifyWithDomain(bytesutil.ToBytes32(msg), pub.val, s.val, bytesutil.ToBytes8(b))
 }
 
-// VerifyAggregate verifies each public key against a message.
+// VerifyAggregate verifies each public key against its respective message.
 // This is vulnerable to rogue public-key attack. Each user must
 // provide a proof-of-knowledge of the public key.
-func (s *Signature) VerifyAggregate(pubKeys []*PublicKey, msg []byte, domain uint64) bool {
+func (s *Signature) VerifyAggregate(pubKeys []*PublicKey, msg [][32]byte, domain uint64) bool {
+	if len(pubKeys) == 0 {
```

Comment:
we can also assert pubkeys is the same length of msg

Reply:
the phore library asserts that over here https://github.com/phoreproject/bls/blob/master/g1pubs/bls.go#L295

Reply:
thanks